registry/config.go

```diff
@@ -379,10 +380,38 @@ func ParseIndexInfo(reposName string) (*IndexInfo, error) {
 
 // NormalizeLocalName transforms a repository name into a normalize LocalName
 // Passes through the name without transformation on error (image id, etc)
+// It does not use the repository info because we don't want to load
+// the repository index and do request over the network.
 func NormalizeLocalName(name string) string {
-	repoInfo, err := ParseRepositoryInfo(name)
+	indexName, remoteName, err := loadRepositoryName(name, true)
 	if err != nil {
 		return name
 	}
-	return repoInfo.LocalName
+
+	var officialIndex bool
+	// Return any configured index info, first.
+	if _, ok := emptyServiceConfig.IndexConfigs[indexName]; ok {
+		officialIndex = true
```

Comment:
could we do
```
if index, ok := emptyServiceConfig.IndexConfigs[indexName]; ok {
  officialIndex = index.Official
}
```

I know its same for `emptyServiceConfig` but I think it's but more understandable.